### PR TITLE
Experiment with contentOnly templateLock on all patterns

### PR DESCRIPTION
**Pull request by noisysocks**
Quick experiment to see how it feels if we add `templateLock: 'contentOnly'` to all patterns.

### CODE REVIEW

File: packages/block-editor/src/components/inserter/menu.js [hunk 1]
```diff
@@ -122,6 +123,13 @@ function InserterMenu(
 
 	const onInsertPattern = useCallback(
 		( blocks, patternName ) => {
+			if ( blocks.length === 1 && blocks[ 0 ]?.name === 'core/group' ) {
```
Comment (getdave) [inline packages/block-editor/src/components/inserter/menu.js:126]: @MaggieCabrera @ndiego I wonder if this is a technique we could use as a proxy for "section pattern"?

Comment (ndiego) [reply packages/block-editor/src/components/inserter/menu.js:126]: Yeah perhaps. You might want to include Cover and maybe Media & Text and Columns.

Comment (MaggieCabrera) [reply packages/block-editor/src/components/inserter/menu.js:126]: We have a core pattern that is just a cover block, and I think TT4 has one of those too

Comment (noisysocks) [reply packages/block-editor/src/components/inserter/menu.js:126]: Worth noting that not all blocks support `templateLock`.

Comment (getdave) [reply packages/block-editor/src/components/inserter/menu.js:126]: If that's the case then we'd need to iterate through the blocks applying the correct blockEditingMode. Template Lock works well for a prototype but possible not for production ready code.

Comment (getdave) [reply packages/block-editor/src/components/inserter/menu.js:126]: I've realised that we're actually looking to find the block that is acting as **the root** of all the blocks we conceptually consider to be "sections" on a template/post.

As a result I think it's important that we review https://github.com/WordPress/gutenberg/pull/64544 which might give us a standardised way to compute this.

**Pull request closed without merging**